### PR TITLE
[YUNIKORN-895] Update dependencies to latest master versions

### PR DESCRIPTION
**Pull request by craigcondit**
### What is this PR for?
Update dependencies in master for core and scheduler-interface. This is necessary for 0.12 as well as to fix e2e test failures.

### What type of PR is it?
* [ ] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [x] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-895

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -22,8 +22,8 @@ go 1.15
 
 require (
 	github.com/GoogleCloudPlatform/spark-on-k8s-operator v0.0.0-20201215015655-2e8b733f5ad0
-	github.com/apache/incubator-yunikorn-core v0.0.0-20211207190138-7ccbebf8230d
-	github.com/apache/incubator-yunikorn-scheduler-interface v0.11.1-0.20211103035638-29e980cee5f1
+	github.com/apache/incubator-yunikorn-core v0.0.0-20211213184834-b34f64e9fd32
+	github.com/apache/incubator-yunikorn-scheduler-interface v0.12.0
```
Comment (yuchaoran2011) [inline go.mod:26]: I'm going to re-tag the latest commit in SI as 0.12.1 for consistency, but it's going to be identical to the previous 0.12.0

Comment (craigcondit) [reply go.mod:26]: Ok, will update PR once we have that. Thanks!

Comment (yuchaoran2011) [reply go.mod:26]: @craigcondit The 0.12.1 tags for both the core and SI have been created. Feel free to use those here

**Comment by craigcondit** [comment]
@yuchaoran2011 can you review?

**Review by yuchaoran2011 — APPROVED** [review]
LGTM. @craigcondit Could you also commit this change to branch-0.12?

**Comment by craigcondit** [comment]
Yes, I will commit to both branches shortly.

**Pull request closed without merging**